gem/spec/carthage_spec.rb

```diff
@@ -0,0 +1,48 @@
+#
+#  Copyright 2016 Google Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+require_relative 'spec_helper'
+
+describe 'carthage' do
+  NIL_YAML = "--- \n...\n".freeze
+
+  it 'successfully modifies the xcode project' do
+    xcodeproj_2 = File.expand_path(File.join(__dir__, 'fixtures', 'carthage_after', 'Example.xcodeproj'))
+
+    Dir.mktmpdir do |tmp_dir|
+      FileUtils.cp_r carthage_before, tmp_dir
+
+      Dir.chdir tmp_dir do
+        # carthage modification of xcodeproj is non-deterministic so we can't rely on
```

Comment:
> This increases the size of the EarlGrey repo.

I'm happy to submit the gem to a new repo if size is an issue.

> Sorry if I'm missing something here, but couldn't we search for specific things like files added or changes in the project file instead of having two projects present for testing?

No. The reason is we need to accomplish two goals with the test: 1) verify the expected changes were made 2) verify nothing else changed.  That's why the [xcodeproj gem](https://github.com/CocoaPods/Xcodeproj/tree/master/spec/fixtures) uses actual projects for tests. 

If we dropped diffing in favor of only looking for expected changes, then it's easy to introduce regressions.